### PR TITLE
Update versions, build on GitHub Actions

### PR DESCRIPTION
**Pull request by camscale**
Update the Centos 8 Dockerfile so that it works as the yum archive has
been moved to `vault.centos.org`.

Add a new Debian 12 Dockerfile as the Debian 8 one no longer builds due
to the signing key of that Debian release having expired.

Pin fpm version in the Dockerfiles to 1.15.1.

Add a GitHub Actions workflow to build and push the centos8 and debian12
images, and remove the `.drone.yml` Drone config.

Issue: https://github.com/gravitational/teleport/issues/20729

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -8,9 +8,62 @@
...
+          echo "V=$V" >> "$GITHUB_ENV"
+      - name: Assume AWS ECR Push Role
+        uses: aws-actions/configure-aws-credentials@010d0da01d0b5a38af31e9c3470dbfdabdecca3a # v4.0.1
+        with:
+          aws-region: us-west-2
+          role-to-assume: arn:aws:iam::146628656107:role/fpm-oci-tag-build-gha
+          role-session-name: "gha-fpm-build=${{ github.run_number }}"
+          role-duration-seconds: 900
+      - name: Log into ECR
+        uses: docker/login-action@v3
```
Comment (wadells) [inline .github/workflows/build.yaml]: Can we pin this too?  I've been moving everything but `actions/`, `github/`, `gravitational/` and `./` (local repo) references to pinning.

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -8,9 +8,62 @@
...
+        with:
+          registry: public.ecr.aws
+        env:
+          AWS_REGION: us-east-1
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@f95db51fddba0c2d1ec667646a06c2ce06100226 # v3.0.0
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      - name: Build FPM Image
+        uses: docker/build-push-action@v3
```
Comment (wadells) [inline .github/workflows/build.yaml]: pin, please

File: debian/12/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+RUN set -x \
+	&& apt-get update && apt-get install -y --no-install-recommends \
+		ruby \
+		ruby-dev \
+		gcc \
+		make \
+		ca-certificates \
+		libffi-dev \
+		ruby-ffi \
```
Comment (wadells) [inline debian/12/Dockerfile:13]: Lets clean up after the apt-get (similar to the `yum clean all` above)

https://docs.docker.com/develop/develop-images/instructions/#apt-get

File: debian/12/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,19 @@
+FROM debian:12
```
Comment (wadells) [inline debian/12/Dockerfile]: Would it make sense to use the `12-slim` tag here, or is that too much of a change?

File: .github/workflows/build.yaml [hunk 5]
```diff
@@ -8,9 +8,62 @@
...
+      contents: read
     steps:
-      - name: Greeting
-        run: echo "This is a dummy build to placate GitHub."
+      - name: version
+        env:
+          VERSION: ${{ inputs.version }}
+          REF_NAME: ${{ github.ref_name }}
+        run: |
+          V="${VERSION:-${REF_NAME}}"
```
Comment (wadells) [inline .github/workflows/build.yaml:38]: I'm not sure `REF_NAME` is safe.  I think git allows stuff like `/` or `+` that docker registries will choke on.

Probably not a problem, as I don't expect these to show up in tag names even if they're technically legal.

**Review by wadells — APPROVED** [review]
I think the overall approach here looks good.  CentOS 8 as our RPM publishing platform makes sense given the options available.

File: debian/12/Dockerfile (hunk 3 above)
Comment (camscale) [reply debian/12/Dockerfile:13]: Done

File: .github/workflows/build.yaml (hunk 1 above)
Comment (camscale) [reply .github/workflows/build.yaml]: Done

File: .github/workflows/build.yaml (hunk 2 above)
Comment (camscale) [reply .github/workflows/build.yaml]: Done, and updated to v5!

File: debian/12/Dockerfile (hunk 4 above)
Comment (camscale) [reply debian/12/Dockerfile]: I've done this. I'll do a test build and ensure it still works

File: .github/workflows/build.yaml (hunk 5 above)
Comment (camscale) [reply .github/workflows/build.yaml:38]: I've fixed this to replace all invalid chars with hyphens.

**Comment by camscale** [comment]
Running a test of the release-linux workflow that uses the latest images I've built with the current commit in this PR. Should be done shortly, which will verify if debian slim works fine.

**Comment by camscale** [comment]
Works fine with debian slim.

**Pull request merged**